src/test/java/com/flightstats/hub/metrics/StatsDFilterTest.java

```diff
@@ -151,4 +151,67 @@ public void testStatsdFilterIsSecondaryReporting_handleNullAndTrue() {
         StatsDFilter statsDFilter = new StatsDFilter(metricsConfig, channelConfigDao, webhookDao);
         assertTrue(statsDFilter.isSecondaryReporting("foo"));
     }
+
+    @Test
+    @SuppressWarnings("unchecked")
+    public void testStatsdFilterExtractName_handleNullTags() {
```

Comment:
I don't know if it's normal, but I usually favor using checked exceptions for test methods rather than suppressing unchecked ones.

Reply:
the cast of the LowerCasedCacheDaoDao<ChannelConfig> and CacheDao<webhook> (mocked) to their interface Dao implementation causes an unchecked warning, totally safe in this context, but I'm happy to change it for consistency's sake.

Reply:
Ooh, yeah. this is fine. I guess you could hide that ugliness in some methods (`getWebhookDao` and `getChannelConfigDao`) that suppress warnings and then the tests wouldn't have to worry about it.

Unrelatedly the imbalance of Channel*Config*Dao and WebhookDao (without the Config) really bugs me. :(

Reply:
I'm not sure my original suggestion actually made sense, tbh. It was a long day.

Reply:
> Unrelatedly the imbalance of ChannelConfigDao and WebhookDao (without the Config) really bugs me. :(

So much!